src/test/java/uk/gov/companieshouse/api/accounts/controller/NoteControllerTest.java

```diff
@@ -0,0 +1,351 @@
+                ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        when(apiResponseMapper.getErrorResponse()).thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse =
+                noteController.update(note, bindingResult,
+                        COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
+        assertNull(responseEntity.getBody());
+    }
+
+    @Test
+    @DisplayName("Get Note resource - success")
+    void getNoteResourceSuccess() throws DataException {
+
+        when(accountsNoteConverter.getAccountsNote(accountType, noteType)).thenReturn(accountingNoteTypeWithExplicitValidation);
+        when(request.getAttribute(AttributeName.TRANSACTION.getValue())).thenReturn(transaction);
+
+        ResponseObject<Note> responseObject = new ResponseObject<>(ResponseStatus.FOUND,
+                note);
+
+        when(noteService.find(accountingNoteTypeWithExplicitValidation, COMPANY_ACCOUNTS_ID))
+                .thenReturn(responseObject);
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.FOUND)
+                .body(responseObject.getData());
+        when(apiResponseMapper.mapGetResponse(responseObject.getData(), request))
+                .thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse =
+                noteController.get(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.FOUND, responseEntity.getStatusCode());
+        assertEquals(note, responseEntity.getBody());
+    }
+
+    @Test
+    @DisplayName("Get Note resource - data exception thrown")
+    void getNoteResourceDataException() throws DataException {
+
+        when(request.getAttribute(AttributeName.TRANSACTION.getValue())).thenReturn(transaction);
+        when(accountsNoteConverter.getAccountsNote(accountType, noteType)).thenReturn(accountingNoteTypeWithExplicitValidation);
+
+        DataException dataException = new DataException("");
+        when(noteService.find(accountingNoteTypeWithExplicitValidation, COMPANY_ACCOUNTS_ID)).thenThrow(dataException);
+
+        ResponseEntity responseEntity =
+                ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+
+        when(apiResponseMapper.getErrorResponse()).thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse = noteController.get(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
+        assertNull(responseEntity.getBody());
+    }
+
+    @Test
+    @DisplayName("Delete Note resource - success")
+    void deleteNoteResourceSuccess() throws DataException {
+
+        when(request.getAttribute(anyString())).thenReturn(transaction);
```

Comment:
Please keep this consistent across methods .
when(request.getAttribute(AttributeName.TRANSACTION.getValue()))

Reply:
As above